org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/SemanticTokensTest.java

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Avaloq Group Ltd (http://www.avaloq.com).
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.ide.tests.server;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.xtext.testing.SemanticTokensFullConfiguration;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * @author Rub�n Porras Campo - Initial test
+ */
+public class SemanticTokensTest extends AbstractTestLangLanguageServerTest {
+	@Test
+	public void testSemanticTokensFull() {
+		testSemanticTokensFull((SemanticTokensFullConfiguration it) -> {
+			String model =
+					"type foo {\n" +
+					"	\n" +
+					"}\n" +
+					"type bar extends foo {\n" +
+					"	\n" +
+					"}\n";
+		it.setModel(model);
+		
+		List<List<Integer>> expectedTokens = new ArrayList<>(); 
+		expectedTokens.add(ImmutableList.of(0,0,4,1,0));
+		expectedTokens.add(ImmutableList.of(3,0,4,1,0));
+		expectedTokens.add(ImmutableList.of(0,9,7,1,0));
+		
+		it.setExpectedText(expectedTokens.stream().flatMap(List::stream).map(i -> i.toString()).collect(Collectors.joining("\n")));
+		});
```

Comment:
Formatting seems to be off.

I find the test very hard to understand. The numbers in isolation are hard to understand. 
Also it seems that the expected text as it is configured would still be ok if I change the expected tokens to a single array of ints since everything is flatmapped into a multiline string. Is that intentional?